### PR TITLE
fix(tests): switch coverage tool, add coveralls

### PR DESCRIPTION
**Pull request by vladikoff**

### CODE REVIEW

**Comment by vladikoff** [comment]
Need to check travis for this one...

**Comment by vladikoff** [comment]
Getting 
```
{"Timestamp":1468958080921000000,"Logger":"fxa-auth-db-server","Type":"bin.metrics.MySql.retryable","Severity":2,"Pid":4398,"EnvVersion":"2.0","Fields":{"err":"OperationalError: getaddrinfo EIO"}}
{"Timestamp":1468958080922000000,"Logger":"fxa-auth-db-server","Type":"bin.metrics.MySql.read","Severity":2,"Pid":4398,"EnvVersion":"2.0","Fields":{"sql":"CALL dbMetadata_1(?)","id":"[\"schema-patch-level\"]","err":"OperationalError: getaddrinfo EIO"}}
{"Timestamp":1468958080924000000,"Logger":"fxa-auth-db-server","Type":"bin.metrics.metrics.run","Severity":2,"Pid":4398,"EnvVersion":"2.0","Fields":{"error":"Error: undefined","stack":"\n    at errnoException (dns.js:37:11)\n    at Object.onanswer [as oncomplete] (dns.js:126:16)\n    --------------------\n    at Protocol._enqueue (/home/travis/build/mozilla/fxa-auth-db-mysql/node_modules/mysql/lib/protocol/Protocol.js:141:48)\n    at Protocol.handshake (/home/travis/build/mozilla/fxa-auth-db-mysql/node_modules/mysql/lib/protocol/Protocol.js:52:41)\n    at PoolConnection.connect (/home/travis/build/mozilla/fxa-auth-db-mysql/node_modules/mysql/lib/Connection.js:136:18)\n    at Pool.getConnection (/home/travis/build/mozilla/fxa-auth-db-mysql/node_modules/mysql/lib/Pool.js:48:16)\n    at PoolCluster._getConnection (/home/travis/build/mozilla/fxa-auth-db-mysql/node_modules/mysql/lib/PoolCluster.js:237:13)\n    at PoolNamespace.getConnection (/home/travis/build/mozilla/fxa-auth-db-mysql/node_modules/mysql/lib/PoolNamespace.js:39:11)\n    at PoolCluster.getConnection (/home/travis/build/mozilla/fxa-auth-db-mysql/node_modules/mysql/lib/PoolCluster.js:145:13)\n    at MySql.promisified [as getClusterConnection] (eval at makeNodePromisifiedEval (/home/travis/build/mozilla/fxa-auth-db-mysql/node_modules/bluebird/js/main/promisify.js:199:12), <anonymous>:15:26)\n    at retryable (/home/travis/build/mozilla/fxa-auth-db-mysql/lib/db/mysql.js:9:33304)\n    at MySql.getConnection (/home/travis/build/mozilla/fxa-auth-db-mysql/lib/db/mysql.js:9:31693)\n    at MySql.singleQuery (/home/travis/build/mozilla/fxa-auth-db-mysql/lib/db/mysql.js:9:24810)\n    at MySql.read (/home/travis/build/mozilla/fxa-auth-db-mysql/lib/db/mysql.js:9:29716)\n    at MySql.readFirstResult (/home/travis/build/mozilla/fxa-auth-db-mysql/lib/db/mysql.js:9:28420)\n    at Function.MySql.connect (/home/travis/build/mozilla/fxa-auth-db-mysql/lib/db/mysql.js:9:3886)\n    at Object.run (/home/travis/build/mozilla/fxa-auth-db-mysql/bin/metrics.js:9:2315)\n    at /home/travis/build/mozilla/fxa-auth-db-mysql/test/local/metrics_tests.js:217:24"}}
test/local/metrics_tests.js ......................... 37/38
  run, with mocked queries
  not ok TypeError: Cannot call method 'close' of undefined
```

locally as well, some weird thingy!

**Comment by vladikoff** [comment]
@vbudhram fixed, r?

File: .travis.yml [hunk 1]
```diff
@@ -40,7 +40,10 @@ before_script:
   - grunt nsp --force
 
 script:
-  - npm test
+  # 'npm run test-travis' runs all tests together to collect all coverage into one from mysql and db-server
+  - COVERALLS_REPO_TOKEN=mlOESL8slkePkodJpcvhF1AUKUH3llXi8 npm run test-travis
+  # memory tests are separate to avoid being included into coverage because they run through the same code
+  - npm run test-mem
```
Comment (vbudhram) [inline .travis.yml]: I wonder if it makes sense to ignore coverage from memory db? Seems like it could be useful since developing/testing locally I mostly use memory db.

File: test/local/metrics_tests.js [hunk 2]
```diff
@@ -207,13 +205,7 @@ DB.connect(config)
             }
           }
         }
-        Object.keys(mocks).forEach(function (key) {
```
Comment (vbudhram) [inline test/local/metrics_tests.js]: Curious why was this removed? Was there conflict with coveralls?

Comment (vladikoff) [inline test/local/metrics_tests.js]: There was some trickery involved in getting the old coverage tool and proxyquire working. seems like it works fine everywhere now

**Comment by vbudhram** [comment]
@vladikoff I think it is fair to report coverage to coveralls for mysql implementation only. Running tests locally will still report mem coverage. Other than that, LGTM 👍

**Comment by vladikoff** [comment]
> @vladikoff I think it is fair to report coverage to coveralls for mysql implementation only. Running tests locally will still report mem coverage. Other than that, LGTM 👍

Yeah let's keep an eye on it and see how it works in practice . thanks for the review!
